salt/states/dockerng.py

```diff
@@ -330,6 +331,14 @@ def _compare(actual, create_kwargs, runtime_kwargs):
                                        'new': desired_hosts}})
                     continue
 
+            elif item == 'volumes':
+                difference = set(data).difference(set(actual_data))
+                desired_data = sorted(data)
+                if difference:
+                    ret.update({item: {'old': sorted(actual_data),
+                                       'new': desired_data}})
+                continue
+
             elif isinstance(data, list):
```

Comment:
The previous implementation was correct. What I intent to change is the behaviour.
so please let's talk about this first. Otherwise we will loose our time.